[16.0][IMP] cetmix_tower_server / cetmix_tower_git / cetmix_tower_webhook UI/UX improvements

ivs-cetmix opened the pull request:
- Clean up views
- Add search panels to improve UX

<img width="400" height="242" alt="image" src="https://github.com/user-attachments/assets/69d89efd-10d8-4324-a202-13176011f694" />

Task: 5139

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enhanced search panels across many views: consolidated "Name/Reference" fields and new advanced filters (Action, Access Level, Tags, OS, Servers, Users, Managers) with icons, counters and multi-select where applicable.
  * Added a Partners/Servers view and a "With Servers" filter; server lists shown on partner pages.
  * Updated button labels for Command and Flight Plan actions.

* **Bug Fixes**
  * Release bumped with a fix for lost custom values in flight plans after skipped commands.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
[BOT] post-merge updates

ivs-cetmix committed:
[IMP] cetmix_tower_server UI/UX improvements

- Add search panel to views to make it easier to search for records
- Clean up search filters, remove duplicate filters, adjust filter names
  and priorities based on usage frequency feedback

Task: 5139

ivs-cetmix committed:
[IMP] cetmix_tower_webhook UI/UX improvements

- Add search panel to views to make it easier to search for records
- Clean up search filters, remove duplicate filters, adjust filter names
  and priorities based on usage frequency feedback

Task: 5139

ivs-cetmix committed:
[IMP] cetmix_tower_git UI/UX improvements

- Add search panel to views to make it easier to search for records

Task: 5139

ivs-cetmix commented on cetmix_tower_server/views/res_partner_view.xml:
```diff
@@ -38,10 +46,36 @@
     </field>
   </record>
+
+  <record id="action_cx_cetmix_tower_partner" model="ir.actions.act_window">
+    <field name="name">Partners</field>
+    <field name="type">ir.actions.act_window</field>
+    <field name="res_model">res.partner</field>
+    <field name="context">{'search_default_filter_with_servers': 1}</field>
+    <field name="view_mode">kanban,tree,form,activity</field>
+  </record>
```
You should use the search_default_<filter_name> format to enable filters

Aldeigja commented on cetmix_tower_server/views/cx_tower_variable_value_view.xml:
```diff
@@ -55,6 +55,14 @@
                     name="global"
                     domain="[('is_global', '=', True)]"
                 />
+                <searchpanel>
```
It might be useful to add filter is_global and server_id, server_template_id here

ivs-cetmix replied:
Thank's for comments, here are the concerns:
- Boolean fields cannot be used as filters.
- Sidepanel is usable as long as it remains compact. If we have 50+ servers it's much easier to search for server instead of scrolling an endless list in the panel.
- Same with server templates.

Aldeigja commented on cetmix_tower_git/views/cx_tower_git_repo_views.xml:
```diff
@@ -122,6 +122,20 @@
                         context="{'group_by': 'owner_id'}"
                     />
                 </group>
+                <searchpanel>
```
Filter is_private can also be useful.

ivs-cetmix replied:
Unfortunately boolean fields are not supported.

Aldeigja reviewed (approved):
Functional LGTM

ivs-cetmix committed:
[IMP] cetmix_tower_aws documentation

Update the documentation to use the new "Command" and "Flight Plan" buttons.

Task: 5139

ivs-cetmix committed:
[IMP] cetmix_tower_ovh documentation

Update the documentation to use the new "Command" and "Flight Plan" buttons.

Task: 5139

ivs-cetmix commented on cetmix_tower_server/views/cx_tower_server_view.xml:
```diff
@@ -515,11 +512,45 @@
+                        context="{'group_by': 'manager_ids'}"
                     />
                 </group>
+                <searchpanel>
+                    <field
+                        name="status"
+                        string="Status"
+                        icon="fa-cog"
+                        enable_counters="1"
+                    />
+                    <field
+                        name="tag_ids"
+                        string="Tags"
+                        icon="fa-tags"
+                        select="multi"
+                        enable_counters="1"
+                    />
+                    <field
+                        name="os_id"
+                        string="OS"
+                        icon="fa-server"
+                        enable_counters="1"
+                    />
+                    <field
+                        name="server_template_id"
+                        string="Server Template"
+                        icon="fa-server"
+                        enable_counters="1"
+                    />
+                </searchpanel>
```
many2many fields are not indexable as they are using an intermediate table for storage

ivs-cetmix commented:
/ocabot merge patch

CetmixGitDrone commented:
This PR looks fantastic, let's merge it!
Prepared branch [16.0-ocabot-merge-pr-446-by-ivs-cetmix-bump-patch](https://github.com/cetmix/cetmix-tower/commits/16.0-ocabot-merge-pr-446-by-ivs-cetmix-bump-patch), awaiting test results.

CetmixGitDrone commented:
Congratulations, your PR was merged at 354bbea6a8e0a478c21cc06e421c2c4b1fa90781. Thanks a lot for contributing to cetmix. ❤️
